eoc_journal/completion_api.py

```diff
@@ -0,0 +1,47 @@
+"""
+A client for completion API for downloading course completion data.
+"""
+from django.conf import settings
+
+from .utils import build_jwt_edx_client
+
+
+class CompletionApiClient(object):
+    """
+    Object builds an API client to make calls to the LMS Grades API.
+    """
+
+    if hasattr(settings, 'LMS_ROOT_URL'):
+        API_BASE_URL = settings.LMS_ROOT_URL + '/api/completion-aggregator/v1'
+    else:
+        API_BASE_URL = None
+
+    def __init__(self, user, course_id):
+        """
+        Connect to the REST API.
+        """
+        self.user = user
+        self.course_id = course_id
+        self.expires_in = getattr(settings, 'OAUTH_ID_TOKEN_EXPIRATION', 300)
+        self.client = None
+        self.connect()
+
+    def connect(self):
+        """
+        Connect to the REST API, authenticating with a JWT for the current user.
+        """
+        scopes = ['profile', 'email']
+
+        self.client = build_jwt_edx_client(
+            self.API_BASE_URL, scopes, self.user,
+            self.expires_in, append_slash=True
+        )
+
+    def get_course_completion(self):
+        """
+        Fetches and returns percentage completion from Course API.
+        """
+        data = self.client \
+            .course(self.course_id) \
+            .get(username=self.user.username).json()
```

Comment:
Could you break this into multiple lines instead to avoid the ``\``?
In the Open edX world the max line length is 120.

Reply:
Done.